src/license.jl

```diff
@@ -3,114 +3,77 @@
-function emptyliccheck(lic::Vector{Cint})
-    return lic
-end
-
-function touchlic(path)
-    f = open(path)
-    return close(f)
-end
-
 function get_xpauthpath(xpauth_path = "", verbose::Bool = true)
     XPAUTH = "xpauth.xpr"
-
-    candidates = []
-
-    # user sent the complete path
-    push!(candidates, xpauth_path)
-
-    # user sent directory
-    push!(candidates, joinpath(xpauth_path, XPAUTH))
-
-    # default env (not metioned in manual)
-    if haskey(ENV, "XPAUTH_PATH")
-        xpauth_path = replace(ENV["XPAUTH_PATH"], "\"" => "")
-        push!(candidates, joinpath(xpauth_path, XPAUTH))
-    end
-
-    # default lib dir
-    if haskey(ENV, "XPRESSDIR")
-        xpressdir = replace(ENV["XPRESSDIR"], "\"" => "")
-        push!(candidates, joinpath(xpressdir, "bin", XPAUTH))
+    candidates = [xpauth_path, joinpath(xpauth_path, XPAUTH)]
+    for key in ("XPAUTH_PATH", "XPRESSDIR")
+        if haskey(ENV, key)
+            push!(candidates, joinpath(replace(ENV[key], "\"" => ""), XPAUTH))
+        end
     end
-
-    # user´s lib dir
     push!(candidates, joinpath(dirname(dirname(libxprs)), "bin", XPAUTH))
-
-    for i in candidates
-        if isfile(i)
+    for candidate in candidates
+        if isfile(candidate)
             if verbose && !haskey(ENV, "XPRESS_JL_NO_INFO")
-                @info("Xpress: Found license file $i")
+                @info("Xpress: Found license file $candidate")
             end
-            return i
+            return candidate
         end
     end
-
     return error(
-        "Could not find xpauth.xpr license file. Check XPRESSDIR or XPAUTH_PATH environment variables.",
+        "Could not find xpauth.xpr license file. Set the `XPRESSDIR` or " *
+        "`XPAUTH_PATH` environment variables.",
     )
 end
+
 """
-    userlic(; liccheck::Function = emptyliccheck, xpauth_path::String = "" )
-Performs license chhecking with `liccheck` validation function on dir `xpauth_path`
+    userlic(;
+        liccheck::Function = identity,
+        verbose::Bool = true,
+        xpauth_path::String = ""
+    )
+
+Performs license chhecking with `liccheck` validation function on dir
+`xpauth_path`
 """
 function userlic(;
+    liccheck::Function = identity,
     verbose::Bool = true,
-    liccheck::Function = emptyliccheck,
     xpauth_path::String = "",
 )
-
+    verbose &= !haskey(ENV, "XPRESS_JL_NO_INFO")
     # change directory to reach all libs
     initdir = pwd()
     if isdir(dirname(libxprs))
         cd(dirname(libxprs))
     end
-
     # open and free xpauth.xpr (touches the file to release it)
     path_lic = get_xpauthpath(xpauth_path, verbose)
-    touchlic(path_lic)
-
+    touch(path_lic)
     # pre allocate vars
-    lic = Cint[1]
-    slicmsg = path_lic #xpauth_path == "dh" ? Array{Cchar}(undef, 1024*8) :
-
-    # FIRST call do xprslicense to get BASE LIC
-    Lib.XPRSlicense(lic, slicmsg)
-
-    # convert BASE LIC to GIVEN LIC
-    lic = liccheck(lic)
-
+    license = Cint[1]
```

Comment:
The `emptyliccheck(lic::Vector{Cint})` assumed `Vector{Cint}`. But this will only ever contain a single element right? I guess changing this to `Ref{Cint}()` will break your existing license check functions.

Reply:
I'm okay keeping this. No need to break just because